numba/core/compiler_machinery.py

```diff
@@ -304,7 +304,11 @@ def debug_print(pass_name, print_condition, printable_condition):
                 else:  # CFG level changes rebuild CFG
                     internal_state.func_ir.blocks = transforms.canonicalize_cfg(
                         internal_state.func_ir.blocks)
-
+            # Check the func_ir only has exactly one Scope instance
+            if not legalize_single_scope(internal_state.func_ir.blocks):
+                raise errors.CompilerError(
+                    f"multiple scope in func_ir detected in {pss}",
```

Comment:
As `self.ENFORCING` is always `False` this never runs. Perhaps it's time to add in `numba.config.enforcing` or something?

Reply:
I'm intentionally putting it in something that doesn't always run. This check is still failing sometimes in a few passes.

Reply:
ok, thanks for clarifying, that there is a way to run it for debug is what is important.